src/Files.App/Services/NetworkDrivesService.cs

```diff
@@ -1,73 +1,208 @@
 	{
-		public bool DisconnectNetworkDrive(ILocatableFolder drive)
+		private ObservableCollection<ILocatableFolder> _Drives;
+		/// <inheritdoc/>
+		public ObservableCollection<ILocatableFolder> Drives
+		{
+			get => _Drives;
+			private set => SetProperty(ref _Drives, value);
+		}
+
+		/// <summary>
+		/// Initializes an instance of <see cref="NetworkDrivesService"/>.
+		/// </summary>
+		public NetworkDrivesService()
 		{
-			return NetworkDrivesAPI.DisconnectNetworkDrive(drive.Path);    
+			_Drives = [];
+
+			var networkItem = new DriveItem
+			{
+				DeviceID = "network-folder",
+				Text = "Network".GetLocalizedResource(),
+				Path = Constants.UserEnvironmentPaths.NetworkFolderPath,
+				Type = DriveType.Network,
+				ItemType = NavigationControlItemType.Drive,
+				MenuOptions = new()
+				{
+					IsLocationItem = true,
+					ShowShellItems = true,
+					ShowProperties = true,
+				},
+			};
+
+			networkItem.MenuOptions.ShowEjectDevice = networkItem.IsRemovable;
+
+			lock (_Drives)
+				_Drives.Add(networkItem);
 		}
 
+		/// <inheritdoc/>
 		public async IAsyncEnumerable<ILocatableFolder> GetDrivesAsync()
 		{
 			var networkLocations = await Win32Helper.StartSTATask(() =>
 			{
 				var locations = new List<ShellLinkItem>();
-				using (var nethood = new ShellFolder(Shell32.KNOWNFOLDERID.FOLDERID_NetHood))
+				using (var netHood = new ShellFolder(Shell32.KNOWNFOLDERID.FOLDERID_NetHood))
 				{
-					foreach (var item in nethood)
+					foreach (var item in netHood)
 					{
 						if (item is ShellLink link)
 						{
 							locations.Add(ShellFolderExtensions.GetShellLinkItem(link));
 						}
 						else
 						{
+							if (item is null)
+								continue;
+
 							var linkPath = (string)item.Properties["System.Link.TargetParsingPath"];
 							if (linkPath is not null)
 							{
 								var linkItem = ShellFolderExtensions.GetShellFileItem(item);
-								locations.Add(new ShellLinkItem(linkItem) { TargetPath = linkPath });
+								locations.Add(new(linkItem) { TargetPath = linkPath });
 							}
 						}
 					}
 				}
+
 				return locations;
 			});
 
 			foreach (var item in networkLocations ?? Enumerable.Empty<ShellLinkItem>())
 			{
 				var networkItem = new DriveItem
 				{
-					Text = System.IO.Path.GetFileNameWithoutExtension(item.FileName),
+					Text = SystemIO.Path.GetFileNameWithoutExtension(item.FileName),
 					Path = item.TargetPath,
 					DeviceID = item.FilePath,
 					Type = DriveType.Network,
 					ItemType = NavigationControlItemType.Drive,
-				};
-				networkItem.MenuOptions = new ContextMenuOptions
-				{
-					IsLocationItem = true,
-					ShowEjectDevice = networkItem.IsRemovable,
-					ShowShellItems = true,
-					ShowProperties = true,
+					MenuOptions = new()
+					{
+						IsLocationItem = true,
+						ShowShellItems = true,
+						ShowProperties = true,
+					},
 				};
 
+				networkItem.MenuOptions.ShowEjectDevice = networkItem.IsRemovable;
+
```

Comment:
I don't think this change makes much sense if `ShowEjectDevice` needs to be set separately.

Reply:
Done!